elastic/kibana
# [Response Ops][Actions] Adding `getAll` to UnsecuredActionsClient

## Summary

Adds unsecured version of the `getAll` function that is exposed via the `unsecuredActionsClient` on the actions plugin start contract. This version doesn't expect any user request and doesn't check RBAC. Actions saved objects are accessed via the internal saved objects repository (instead of a saved objects client scoped to the request) and an ES client that uses the internal user (instead of scoped to the request). This function returns all connectors for the give space ID and preconfigured connectors but does not return system actions.

## To Verify
1. Run this PR and create a connector via the UI or API in multiple different spaces.
2. In `x-pack/plugins/alerting/server/plugin.ts` in the start function, add the following:

```
const unsecuredActionsClient = plugins.actions.getUnsecuredActionsClient();
unsecuredActionsClient.getAll('default').then((results) => {
  console.log('default connectors');
  console.log(JSON.stringify(results));
});

unsecuredActionsClient.getAll(<differentSpaceId>).then((results) => {
  console.log(JSON.stringify(results));
});
```
3. Restart Kibana and verify that the appropriate connectors were returned and logged.
